applications/CompressiblePotentialFlowApplication/custom_elements/incompressible_adjoint_potential_flow_element.cpp

```diff
@@ -0,0 +1,410 @@
+            {
+                if(distances[i] < 0)
+                    rResult[NumNodes+i] = GetGeometry()[i].GetDof(ADJOINT_VELOCITY_POTENTIAL).EquationId();
+                else
+                    rResult[NumNodes+i] = GetGeometry()[i].GetDof(ADJOINT_AUXILIARY_VELOCITY_POTENTIAL,0).EquationId();
+            }
+        }
+
+
+    }
+
+    template <class TPrimalElement>
+    void AdjointIncompressiblePotentialFlowElement<TPrimalElement>::GetDofList(DofsVectorType& rElementalDofList, ProcessInfo& CurrentProcessInfo) 
+    {
+        const AdjointIncompressiblePotentialFlowElement& r_this = *this;
+        const int wake = r_this.GetValue(WAKE);
+        const int kutta = r_this.GetValue(KUTTA);
+
+        if(wake == 0) //normal element
+        {
+            if (rElementalDofList.size() != NumNodes)
+                rElementalDofList.resize(NumNodes);
+
+            if(kutta == 0){
+                for (unsigned int i = 0; i < NumNodes; i++)
+                    rElementalDofList[i] = GetGeometry()[i].pGetDof(ADJOINT_VELOCITY_POTENTIAL);
+            }
+            else{
+                for (unsigned int i = 0; i < NumNodes; i++){
+                    if (!GetGeometry()[i].GetValue(TRAILING_EDGE))
+                        rElementalDofList[i] = GetGeometry()[i].pGetDof(ADJOINT_VELOCITY_POTENTIAL);
+                    else
+                        rElementalDofList[i] = GetGeometry()[i].pGetDof(ADJOINT_AUXILIARY_VELOCITY_POTENTIAL);
+                }
+            }
+        }
+        else//wake element
+        {
+            if (rElementalDofList.size() != 2*NumNodes)
+                rElementalDofList.resize(2*NumNodes);
+
+            array_1d<double,NumNodes> distances;
+            GetWakeDistances(distances);
+
+            //positive part
+            for (unsigned int i = 0; i < NumNodes; i++)
+            {
+                if(distances[i] > 0)
+                    rElementalDofList[i] = GetGeometry()[i].pGetDof(ADJOINT_VELOCITY_POTENTIAL);
+                else
+                    rElementalDofList[i] = GetGeometry()[i].pGetDof(ADJOINT_AUXILIARY_VELOCITY_POTENTIAL);
+            }
+
+            //negative part - sign is opposite to the previous case
+            for (unsigned int i = 0; i < NumNodes; i++)
+            {
+                if(distances[i] < 0)
+                    rElementalDofList[NumNodes+i] = GetGeometry()[i].pGetDof(ADJOINT_VELOCITY_POTENTIAL);
+                else
+                    rElementalDofList[NumNodes+i] = GetGeometry()[i].pGetDof(ADJOINT_AUXILIARY_VELOCITY_POTENTIAL);
+            }
+        }
+    }
+
+    template <class TPrimalElement>
+    int AdjointIncompressiblePotentialFlowElement<TPrimalElement>::Check(const ProcessInfo& rCurrentProcessInfo) 
+    {
+
+        KRATOS_TRY
+
+        if (this->Id() < 1)
+        {
+            KRATOS_THROW_ERROR(std::logic_error, "AdjointIncompressiblePotentialFlowElement found with Id 0 or negative", "")
```

Comment:
I think this check is already done in the baseclass => better to call that method

Reply:
Yes! Actually i will change to mpPrimalElement -> Check()

thanks!